recipes/pybind11/all/test_package/conanfile.py

```diff
@@ -21,7 +42,7 @@ def _python_interpreter(self):
         return sys.executable
 
     def test(self):
-        if not tools.cross_building(self):
-            with tools.environment_append({"PYTHONPATH": "lib"}):
-                self.run("{} {}".format(self._python_interpreter, os.path.join(
-                    self.source_folder, "test.py")), run_environment=True)
+        if can_run(self):
+            python_path = os.path.join(self.build_folder, self.cpp.build.libdirs[0])
+            module_path = os.path.join(self.source_folder, "test.py")
+            self.run(f"PYTHONPATH={python_path} {self._python_interpreter} {module_path}", env="conanrun")
```

Comment:
Is there a better way to set this in Conan V2? I can't set it in `generate` as I don't have access to `self.build_folder` at that point.

Reply:
That's a great question. Hopefully someone who knows will update us.